packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -104,114 +128,92 @@ export class MirrorNodeClient {
     }
 
+    public async getContractResultsByAddressAndTimestamp(contractIdOrAddress: string, timestamp: string) {
+        return this.request(`${this.getContractResultsByAddressPath(contractIdOrAddress)}/${timestamp}`, [206, 400, 404]);
+    }
+
     public async getContractResultsLogs(
-        index?: number,
-        limit?: number,
-        order?: string,
-        timestamp?: string | [string],
-        topic0?: string | [string],
-        topic1?: string | [string],
-        topic2?: string | [string],
-        topic3?: string | [string]) {
+        contractLogsResultsParams?: IContractLogsResultsParams,
+        limitOrderParams?: ILimitOrderParams) {
         const queryParamObject = {};
-        this.setQueryParam(queryParamObject, 'index', index);
-        this.setQueryParam(queryParamObject, 'limit', limit);
-        this.setQueryParam(queryParamObject, 'order', order);
-        this.setQueryParam(queryParamObject, 'timestamp', timestamp);
-        this.setQueryParam(queryParamObject, 'topic0', topic0);
-        this.setQueryParam(queryParamObject, 'topic1', topic1);
-        this.setQueryParam(queryParamObject, 'topic2', topic2);
-        this.setQueryParam(queryParamObject, 'topic3', topic3);
+        if (contractLogsResultsParams) {
+            this.setQueryParam(queryParamObject, 'index', contractLogsResultsParams.index);
+            this.setQueryParam(queryParamObject, 'timestamp', contractLogsResultsParams.timestamp);
+            this.setQueryParam(queryParamObject, 'topic0', contractLogsResultsParams.topic0);
+            this.setQueryParam(queryParamObject, 'topic1', contractLogsResultsParams.topic1);
+            this.setQueryParam(queryParamObject, 'topic2', contractLogsResultsParams.topic2);
+            this.setQueryParam(queryParamObject, 'topic3', contractLogsResultsParams.topic3);
+        }
+
+        this.setLimitOrderParams(queryParamObject, limitOrderParams);
         const queryParams = this.getQueryParams(queryParamObject);
         return this.request(`${MirrorNodeClient.GET_CONTRACT_RESULT_LOGS_ENDPOINT}${queryParams}`, [400]);
     }
 
     public async getLatestBlock() {
-        return this.getBlocks(undefined, undefined, 1, MirrorNodeClient.ORDER.DESC);
+        return this.getBlocks(undefined, undefined, this.getLimitOrderQueryParam(1, MirrorNodeClient.ORDER.DESC));
+    }
+
+    public getLimitOrderQueryParam(limit: number, order: string): ILimitOrderParams {
+        return { limit: limit, order: order };
     }
 
     public async getNetworkExchangeRate(timestamp?: string) {
         const queryParamObject = {};
         this.setQueryParam(queryParamObject, 'timestamp', timestamp);
         const queryParams = this.getQueryParams(queryParamObject);
-        return this.request(`${MirrorNodeClient.GET_NETWORK_EXCHANGERATE_ENDPOINT}${queryParams}`, [400]);
+        return this.request(`${MirrorNodeClient.GET_NETWORK_EXCHANGERATE_ENDPOINT}${queryParams}`, [400, 404, 500]);
```

Comment:
Should 5xx be expected error codes?  I thought we didn't want to paper over internal errors?

Reply:
Correct, MirrorNode has it in case the file is malformed but yeah we should need to support that

Reply:
Fixed